### PR TITLE
feat: add udiskie to automatically mount usb mass storages

### PR DESCRIPTION
**Pull request by rubiin**
# Pull Request

## Description

Please read these instructions and remove unnecessary text.

- Try to include a summary of the changes and which issue is fixed.
- Also include relevant motivation and context (if applicable).
- List any dependencies that are required for this change. (e.g., packages or other PRs)
- Provide a link if there is an issue related to this pull request. e.g., Fixes # (issue)
- Please add Reviewers, Assignees, Labels, Projects, and Milestones to the PR. (if applicable)

## Type of change

Please put an `x` in the boxes that apply:

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [ ] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [x] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added a changelog entry.
- [x] I have added necessary comments/documentation to my code.
- [ ] I have added tests to cover my changes.
- [x] I have tested my code locally and it works as expected.
- [ ] All new and existing tests passed.

## Screenshots

(if appropriate)

### CODE REVIEW

File: Scripts/custom_hypr.lst [hunk 1]
```diff
@@ -14,7 +14,7 @@ bluez                                                  # bluetooth protocol stac
 bluez-utils                                            # bluetooth utility cli
 blueman                                                # bluetooth manager gui
 brightnessctl                                          # screen brightness control
-
+udiskie                                                # manage removable media
```
Comment (Skerse) [inline Scripts/custom_hypr.lst:17]: Re-add the empty line please.

**Review by Skerse — CHANGES_REQUESTED** [review]

File: Configs/.config/hypr/hyprland.conf [hunk 2]
```diff
@@ -31,6 +31,7 @@ exec-once = systemctl --user import-environment WAYLAND_DISPLAY XDG_CURRENT_DESK
 exec-once = $scrPath/polkitkdeauth.sh # Authentication dialogue for GUI apps
 exec-once = waybar # Launch the system bar
 exec-once = blueman-applet # Systray app for Bluetooth
+exec-once = udiskie &
```
Comment (prasanthrangan) [inline Configs/.config/hypr/hyprland.conf:34]: this `&` is not required here.

**Comment by prasanthrangan** [comment]
is `udiskie` really necessary as part of core packages? as automount works for me without this.

File: Configs/.config/hypr/hyprland.conf (hunk 2 above)
Comment (rubiin) [reply Configs/.config/hypr/hyprland.conf:34]: this was copied from hyperland docs , so i will remove it

**Comment by rubiin** [comment]
> is `udiskie` really necessary as part of core packages? as automount works for me without this.

wel not required for mount to function but it does provide a gui for mounting stuffs
cc: https://wiki.hyprland.org/Useful-Utilities/Other/
https://github.com/coldfix/udiskie

**Comment by kRHYME7** [comment]
Same concerns too, can we move this to custom apps instead?

**Comment by rubiin** [comment]
> Same concerns too, can we move this to custom apps instead?
> 
> Looks good though. I assume this would likely be the same functionality on Windows mounting USBs?

lets move this to custom apps then

**Comment by prasanthrangan** [comment]
> lets move this to custom apps then, I will open up a new PR

nevermind, my bad... I never knew that `udiskie` provides a sys tray menu :)
please keep your previous commit, its fine!

**Review by Skerse — COMMENTED** [review]
Comments after lines should be lowercase, also add a comment to udiskie to give it's purpose. I would also recommend like others to move this to `custom_apps.lst`, which if we'll be doing that, the autostart with `exec-once` better belongs to `Configs/.config/hypr/userprefs.t2` in the "launch" area.

**Comment by kRHYME7** [comment]
> lets move this to custom apps then, I will open up a new PR 

I think it's good on defaults! Would be very helpful! Sorry for not reading the whole udiskie docs 😅

**Comment by rubiin** [comment]
cool then , i will address some comments and then it will be ready

File: Configs/.config/hypr/hyprland.conf [hunk 3]
```diff
@@ -24,19 +24,20 @@ monitor = ,preferred,auto,auto
...
+exec-once = $scrPath/polkitkdeauth.sh # Authentication dialogue for GUI apps
+exec-once = waybar # Launch the system bar
+exec-once = blueman-applet # Systray app for Bluetooth
+exec-once = udiskie # Front-end that allows to manage removable media
+exec-once = nm-applet --indicator # Systray app for Network/Wifi
+exec-once = dunst # Start notification demon
+exec-once = wl-paste --type text --watch cliphist store # Clipboard store text data
+exec-once = wl-paste --type image --watch cliphist store # Clipboard store image data
+exec-once = $scrPath/swwwallpaper.sh # Start wallpaper daemon
+exec-once = $scrPath/batterynotify.sh # Battery notification
```
Comment (Skerse) [inline Configs/.config/hypr/hyprland.conf:40]: Thanks for adding the comment, but can you make the comments here all start lowercase like it currently is? It's not necessary, it's just for consistency with the rest of the project. I'll approve it when that's done!

Comment (rubiin) [reply Configs/.config/hypr/hyprland.conf:40]: sure thing

**Review by prasanthrangan — APPROVED** [review]

**Pull request merged**